### PR TITLE
[no ticket][risk=no] Puppeteer: refactor workspace pages

### PR DESCRIPTION
**Pull request by aweng98**
* Create new `workspace-base.ts` abstract class for extend by `workspace-data-page.ts`, `workspace-analysis-page.ts` and `workspace-about-page.ts` classes. It contains common workspace-page functions.

* Rename `data-page.ts` to `workspace-data-page.ts` to have a consistent name convention like other `workspace-*.ts` page names.

### CODE REVIEW

File: e2e/app/page/workspace-about-page.ts [hunk 1]
```diff
@@ -32,11 +32,6 @@ export default class WorkspaceAboutPage extends AuthenticatedPage{
     }
   }
 
-  async isOpen(): Promise<boolean> {
```
Comment (aweng98) [inline e2e/app/page/workspace-about-page.ts:35]: Moved it to `workspace-base.ts` file to be a common function.

File: e2e/app/page/workspace-data-page.ts [hunk 2]
```diff
@@ -67,24 +64,6 @@ export default class DataPage extends AuthenticatedPage {
     return ellipsisMenu.clickAction(action);
   }
 
-  /**
-   * Select DATA, ANALYSIS or ABOUT page tab.
-   * @param {TabLabelAlias} tabName
-   * @param opts
-   */
-  async openTab(tabName: TabLabelAlias, opts: {waitPageChange?: boolean} = {}): Promise<void> {
```
Comment (aweng98) [inline e2e/app/page/workspace-data-page.ts:75]: Moved to `workspace-base.ts` file to be a common function.

File: e2e/app/page/workspace-base.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
+import Link from 'app/element/link';
+import {buildXPath} from 'app/xpath-builders';
+import {ElementType} from 'app/xpath-options';
+import {Page} from 'puppeteer';
+import {waitWhileLoading} from 'utils/test-utils';
+import {waitForAttributeEquality} from 'utils/waits-utils';
+import AuthenticatedPage from './authenticated-page';
+
+export enum TabLabelAlias {
```
Comment (ericsong) [inline e2e/app/page/workspace-base.ts:9]: nit: I think `TabLabels` is shorter and more descriptive. or `WorkspaceTabs`

File: e2e/tests/workspace/workspace-create.spec.ts [hunk 4]
```diff
@@ -12,7 +12,7 @@ describe('Creating new workspaces', () => {
     await signIn(page);
   });
 
-  test('Create workspace - NO request for review', async () => {
+  test.skip('Create workspace - NO request for review', async () => {
```
Comment (ericsong) [inline e2e/tests/workspace/workspace-create.spec.ts:15]: is this test broken now?

File: e2e/app/page/workspace-base.ts (hunk 3 above)
Comment (aweng98) [reply e2e/app/page/workspace-base.ts]: 👍 Tabs also exists in Home page. `TabLabels` is better.

File: e2e/tests/workspace/workspace-create.spec.ts (hunk 4 above)
Comment (aweng98) [reply e2e/tests/workspace/workspace-create.spec.ts]: good catch. I forgot to un-skip after local testing.

**Pull request merged**